From df20c6b52c6576832a1bc0ac53c53695c4dcacb0 Mon Sep 17 00:00:00 2001
From: vishalpowar <vishalpowar@google.com>
Date: Wed, 30 Mar 2022 14:07:55 -0700
Subject: Update gcp mesh-name to 'td.networking.gke.io/mesh-name' (#38185)

* Changes to make pilot-agent generate correct bootstrap to connect to GCP
TrafficDirector

The changes in the PR fall in two categories
1) bootstrap template (gcp_envoy_bootstrap.json)
  - Change to support the new xDS V3 nodeID format used by
  TrafficDirector
  - Change to explictly support V3 API when connecting to
  TrafficDirector
  - Changes to support istio-proxy inbound interception.

2) supporting changes in pilot-agent to set bootstrap template
varaibles.

cla: yes

* Pass the metadata.Labels to the bootstrap template and change
gcp_envoy_bootstrap file to use information to generate correct NodeId

* Pass the metadata instead of just label to the template.

Also, update the gcp_envoy_bootstrap label lookup from "gke.io/mesh" to "td.io/scope"

* Remove the metadata entries for TRAFFICDIRECTOR_ as this will come from
sidecar injector

* Update the gcp_envoy_bootstrap.json to use correct label for looking up
mesh-name

* Update fcp_envoy_bootstrap.json to use the correct label 'networking.gke.io/mesh/mesh-name'

* Update gcp mesh-name to 'td.networking.gke.io/mesh-name'
---
 tools/packaging/common/gcp_envoy_bootstrap.json | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/packaging/common/gcp_envoy_bootstrap.json b/tools/packaging/common/gcp_envoy_bootstrap.json
index 52b5e9c966..0b28495a43 100644
--- a/tools/packaging/common/gcp_envoy_bootstrap.json
+++ b/tools/packaging/common/gcp_envoy_bootstrap.json
@@ -1,7 +1,7 @@
 {
   "node": {
-    {{ if (index .metadata.Labels "networking.gke.io/mesh/mesh-name") }}
-      "id": "projects/{{ .gcp_project_number }}/networks/{{ (index .metadata.Labels "networking.gke.io/mesh/mesh-name") }}/nodes/{{ .nodeID}}",
+    {{ if (index .metadata.Labels "td.networking.gke.io/mesh-name") }}
+      "id": "projects/{{ .gcp_project_number }}/networks/{{ (index .metadata.Labels "td.networking.gke.io/mesh-name") }}/nodes/{{ .nodeID}}",
     {{ else }}
       "id": "projects/{{ .gcp_project_number }}/networks/default/nodes/{{ .nodeID}}",
     {{ end }}
-- 
2.35.3

